autoload/arduino.vim

```diff
@@ -380,7 +439,21 @@ function! arduino#Upload() abort
     call slime#send(cmd."\r")
   elseif !empty($TMUX) && !empty(g:arduino_upload_tmux)
     " the call to $SHELL -i is to prevent the split to directly close after the command has finished
-    exe "silent exec \"!tmux " . g:arduino_upload_tmux . " '" . cmd . "; $SHELL -i'\""
+    call arduino#TmuxRunCommand(g:arduino_upload_tmux, cmd, get(g:, 'arduino_upload_tmux_pane_title', -1))
+  else
+    exe s:TERM . cmd
+  endif
+  return v:shell_error
+endfunction
+
+
+function! arduino#UploadUsingProgrammer() abort
+  let cmd = arduino#GetArduinoCommand("--upload --useprogrammer")
+  if g:arduino_use_slime
+    call slime#send(cmd."\r")
+  elseif !empty($TMUX) && !empty(g:arduino_upload_tmux)
+    " the call to $SHELL -i is to prevent the split to directly close after the command has finished
+    call arduino#TmuxRunCommand(g:arduino_upload_tmux, cmd, get(g:, 'arduino_upload_tmux_pane_title', -1))
```

Comment:
I think it makes more sense to make the `--useprogrammer` option a global variable instead of creating a new function for it. It'll fit right in to the `arduino#Upload` function.

Reply:
Changes included. Use programmer is forced when `g:arduino_upload_using_programmer` is set to 1, by default equals 0.